tests/Services/SearchServiceTest.php

```diff
@@ -155,6 +156,177 @@ public function testRedirectToProductPageOnDoSearch(
+        array $attributes
+    ) {
+        $this->client->expects($this->any())->method('call')->willReturn(Plugin::API_ALIVE_RESPONSE_BODY);
+
+        $responseMock = $this->getMockBuilder(Response::class)->disableOriginalConstructor()->getMock();
+        $responseMock->expects($this->once())->method('getVariationIds')->willReturn($responseVariationIds);
+        $this->responseParser->expects($this->once())->method('parse')->willReturn($responseMock);
+
+        $itemSearchServiceMock = $this->getMockBuilder(ItemSearchService::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $itemSearchServiceMock->expects($this->at(0))
+            ->method('getResult')
+            ->willReturn($itemSearchServiceResultsAll);
+
+        $searchServiceMock = $this->getSearchServiceMock([
+            'getCategoryService',
+            'getItemSearchService',
+            'getSearchFactory',
+            'handleProductRedirectUrl'
+        ]);
+        $searchServiceMock->expects($this->any())
+            ->method('getItemSearchService')
+            ->willReturn($itemSearchServiceMock);
+
+        $expectedCategoryId = 1234;
+        $this->configRepository->expects($this->once())->method('get')
+            ->with('IO.routing.category_search')
+            ->willReturn($expectedCategoryId);
+
+        $categoryMock = $this->getMockBuilder(Category::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $categoryMock->id = $expectedCategoryId;
+
+        $categoryServiceMock = $this->getMockBuilder(CategoryService::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $categoryServiceMock->expects($this->any())
+            ->method('getCurrentCategory')
+            ->willReturn($categoryMock);
+
+        $searchServiceMock->expects($this->any())
+            ->method('getCategoryService')
+            ->willReturn($categoryServiceMock);
+
+        /** @var ExternalSearch|MockObject $externalSearchMock */
+        $externalSearchMock = $this->getMockBuilder(ExternalSearch::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['setResults'])
+            ->getMock();
+        $externalSearchMock->categoryId = $expectedCategoryId;
+
+        $externalSearchMock->expects($this->once())->method('setResults');
+
+        /** @var HttpRequest|MockObject $requestMock */
+        $requestMock = $this->getMockBuilder(HttpRequest::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $requestMock->expects($this->any())
+            ->method('all')
+            ->willReturn($attributes);
+
+        /** @var Request|HttpRequest|MockObject $searchRequestMock */
+        $searchRequestMock = $this->getMockBuilder(Request::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        // Ensure that the given category is null. This way we are sure that the category is being ignored.
```

Comment:
But what about scenario with `'cat' => 'Blubbergurken'`? Can you test this category isn't ignored?